### ISSUE TITLE
Digital ocean driver create command fails on new ubuntu 16.04 LTS image

### ISSUE THREAD

**Issue by raarts** [docker/machine#3358]
```
$ docker-machine create -d digitalocean --digitalocean-access-token xxx --digitalocean-ssh-key-fingerprint xxx --digitalocean-image ubuntu-16-04-x64  test-machine
Running pre-create checks...
Creating machine...
(test-machine) Creating SSH key...
(test-machine) Creating Digital Ocean droplet...
(test-machine) Waiting for IP address to be assigned to the Droplet...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Error creating machine: Error running provisioning: Something went wrong running an SSH command!
command : sudo apt-get update
err     : exit status 100
output  : Reading package lists...
E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/
```

**Comment by nathanleclaire** [comment]
Hrm, I can dupe.

**Comment by nathanleclaire** [comment]
`docker-machine provision node` seems to fix it for me, so I'm guessing we have our typical race-due-to-shared-apt-properties problem here.

**Comment by nathanleclaire** [comment]
Hey @tianon, curious what your opinion is here.  Think we should maybe retry the `apt-get update` a few times with a sleep in case someone else is racing us to use the lock?

**Comment by nathanleclaire** [comment]
(It seems likely this issue is likely related to some kind of node bootstrap process DO is running behind the scenes)

**Comment by tianon** [comment]
Whoa, that's strange -- any idea what the DO bootstrap is doing?  Maybe just `apt-get update && apt-get dist-upgrade`?  Do we have any DO contacts that could tell us whether this is expected and/or whether we can inhibit it in some way?  Just retrying feels really hacky. :disappointed:

**Comment by nathanleclaire** [comment]
> Whoa, that's strange -- any idea what the DO bootstrap is doing? Maybe just apt-get update && apt-get dist-upgrade? Do we have any DO contacts that could tell us whether this is expected and/or whether we can inhibit it in some way? Just retrying feels really hacky.

Yeah, I know right?

I think that there's some kind of `cloud-init` thing running, although I'm not sure why they would be doing package operations without request.  Reaching out to DO team seems a decent idea.

**Comment by stunti** [comment]
Would it be possible to have a flag to prevent docker-machine from running "apt-get update" ?

**Comment by raarts** [comment]
Why not wait for /var/lib/apt/lists/lock to disappear?

**Comment by nathanleclaire** [comment]
> Why not wait for /var/lib/apt/lists/lock to disappear?

Just off the top of my head, it's a bit fragile as a solution.  You have no real guarantee that's going to disappear in any reasonable amount of time, if an `apt` process aborts in the middle it will still be around and never get cleaned up, forcing us to introduce additional timeouts around this, and so on.  It would be far better to handle in `apt` itself if possible, and/or establish what exactly DO are doing behind the scenes that's causing this before implementing anything to try and fix it.

**Comment by ssteinerx** [comment]
Just got this from DO support:

As for the issue, I went ahead and reported this up to our Images team for review, but in the meantime we would recommend just adding a sleep after creation, as our API returns the droplet as "built" once the create process is done, but this does not include the time it can take for cloud-init to run, which normally takes less than 30 seconds, so the best work around currently would be to use a sleep. You can also watch the log at /var/log/cloud-init.log as this can be used to track when the provisioning process is complete with the droplet. 

As for the apt-get lock, I don't have any feedback from our images team yet as to why this is a part of the cloud-init process, so this is why we recommend the above workaround until I get more information on our side as to why this is occurring.

**Comment by utlemming** [comment]
Thanks for your patience...apt seems to be leaving a dangling /var/lib/apt/lock. 
```

ben@prongs:~$ docker-machine create -d digitalocean 
      --digitalocean-access-token <REDACTED> 
      --digitalocean-image ubuntu-16-04-x64  test-machine3
Running pre-create checks...
Creating machine...
(test-machine3) Creating SSH key...
(test-machine3) Creating Digital Ocean droplet...
(test-machine3) Waiting for IP address to be assigned to the Droplet...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Installing Docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Checking connection to Docker...
Docker is up and running!
To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env test-machine3

```

**Comment by ssteinerx** [comment]
From DO support and I just confirmed IT WORKS!

Thanks for waiting! I spoke with our engineers, and they think they found the cause of the issue, and updated our Ubuntu 16.04 images. Can you please test to see if the issue is still occuring when you use Docker to deploy on our 16.04 image? 

As for the issue, it sounds like there may have been a bug with cloud-init where apt was leaving a lock file after being ran, which caused the issue.

**Comment by nathanleclaire** [comment]
Thanks @ssteinerx !  I'm going to go ahead and close.

**Comment by ssteinerx** [comment]
Crap, first thing this morning...

./create-do machine
Creating:  machine
Running pre-create checks...
Creating machine...
(machine) Creating SSH key...
(machine) Creating Digital Ocean droplet...
(machine) Waiting for IP address to be assigned to the Droplet...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Error creating machine: Error running provisioning: Something went wrong running an SSH command!
command : sudo apt-get update
err     : exit status 100
output  : Reading package lists...
E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/

**Comment by ssteinerx** [comment]
Failed twice in a row.  Fixed-ish yesterday, borken today.  Reported to DO support.

**Comment by jjasonclark** [comment]
Still broken as of May 27th 2016 in the same place

**Comment by ssteinerx** [comment]
Please reopen, this is still broken.

**Comment by arianitu** [comment]
I have the same issue

**Comment by randunel** [comment]
Still broken, please reopen.

**Comment by bamarni** [comment]
@randunel : this was closed because it's a digitalocean issue, please report it to their support instead.

**Comment by cadavre** [comment]
But it is not only DO issue. Same thing happen with AWS driver, but after machine is created you can ssh into and run given command manually – now without errors, because it looks like lock file is gone. This shall be solved on Docker Machine side, i.e. by adding some retries or sleep time to give cloud instances enough time to make cloud-init.

This is also a duplicate of #2890.

**Comment by rvion** [comment]
it happens half the time on my side 😢 

```
Provisioning with ubuntu(systemd)...
Error creating machine: Error running provisioning: Error running apt-get update: ssh command error:
command : sudo apt-get update
err     : exit status 100
output  : Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [102 kB]
Get:5 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages [355 kB]
Get:6 http://security.ubuntu.com/ubuntu xenial-security/main Translation-en [155 kB]
Get:7 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 Packages [167 kB]
Get:8 http://security.ubuntu.com/ubuntu xenial-security/universe Translation-en [87.2 kB]
Get:9 http://security.ubuntu.com/ubuntu xenial-security/multiverse amd64 Packages [2,752 B]
Get:10 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages [632 kB]
Get:11 http://archive.ubuntu.com/ubuntu xenial-updates/main Translation-en [261 kB]
Get:12 http://archive.ubuntu.com/ubuntu xenial-updates/restricted amd64 Packages [8,048 B]
Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/restricted Translation-en [2,688 B]
Get:14 http://archive.ubuntu.com/ubuntu xenial-updates/universe amd64 Packages [529 kB]
Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/universe Translation-en [210 kB]
Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 Packages [15.5 kB]
Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/multiverse Translation-en [7,540 B]
Get:18 http://archive.ubuntu.com/ubuntu xenial-backports/main amd64 Packages [4,884 B]
Get:19 http://archive.ubuntu.com/ubuntu xenial-backports/main Translation-en [3,216 B]
Get:20 http://archive.ubuntu.com/ubuntu xenial-backports/universe amd64 Packages [5,800 B]
Get:21 http://archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [3,004 B]
Reading package lists...
E: can not open /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_InRelease - fopen (2: No such file or directory)
```

**Comment by lukepolo** [comment]
i have to delay for 45 seconds

**Comment by scp93ch** [comment]
I've had a similar problem on OpenStack when I want to run ``apt-get update`` on a newly instantiated compute instance. OpenStack also runs cloud-init as soon as a machine is booted.

The problem is that if you run ``apt-get update`` before cloud-init runs then you get an out of date ``/etc/apt/sources.list``. If you run it during cloud-init then you can get a partial file in ``sources.list`` which causes several odd errors. You have to wait for cloud-init to finish.

I added the following to my provisioning script:

```shell
# wait for cloud-init to complete (it updates the /etc/apt/sources.list file)
while ssh ${SSH_ARGS} ${IMAGE_USER}@${IP} "ps aux | grep cloud-init | grep -v grep > /dev/null"; do
    echo "Waiting for cloud-init to complete..."
    sleep 10
done
```

Obviously, replace the SSH_ARGS etc with what you need.

**Comment by lukepolo** [comment]
This worked perfect for me although I changed it to be a bit different because I queue things 


`ps aux | grep $service | grep -v grep | wc -l`
